embrace-android-sdk/src/androidTest/assets/golden-files/session-end.json

```diff
@@ -1,167 +1,262 @@
     },
     {
-      "attributes": {
-        "emb.sequence_id": "3",
-        "emb.key": "true",
-        "emb.type": "performance"
+      "attributes":{
+        "emb.sequence_id":"3",
+        "emb.key":"true",
+        "emb.type":"performance"
       },
-      "end_time_unix_nano": "__EMBRACE_TEST_IGNORE__",
-      "events": [],
-      "name": "emb-startup-moment",
-      "parent_span_id": "0000000000000000",
-      "span_id": "__EMBRACE_TEST_IGNORE__",
-      "start_time_unix_nano": "__EMBRACE_TEST_IGNORE__",
-      "status": "OK",
-      "trace_id": "__EMBRACE_TEST_IGNORE__"
+      "end_time_unix_nano":"__EMBRACE_TEST_IGNORE__",
+      "events":[
+
+      ],
+      "name":"emb-startup-moment",
+      "parent_span_id":"0000000000000000",
+      "span_id":"__EMBRACE_TEST_IGNORE__",
+      "start_time_unix_nano":"__EMBRACE_TEST_IGNORE__",
+      "status":"OK",
+      "trace_id":"__EMBRACE_TEST_IGNORE__"
     },
     {
-      "attributes": {
-        "emb.sequence_id": "1",
-        "emb.private": "true",
-        "emb.type": "ux.session",
-        "emb.okhttp3": "true",
-        "emb.okhttp3_on_classpath": "4.9.3",
-        "emb.kotlin_on_classpath": "1.4.32",
-        "emb.is_emulator": "__EMBRACE_TEST_IGNORE__",
-        "emb.usage.add_breadcrumb": "1",
-        "emb.usage.add_user_persona": "1",
-        "emb.usage.set_user_email": "1",
-        "emb.usage.set_user_identifier": "1",
-        "emb.usage.set_username": "1"
+      "attributes":{
+        "emb.sequence_id":"1",
+        "emb.private":"true",
+        "emb.type":"ux.session",
+        "emb.okhttp3":"true",
+        "emb.okhttp3_on_classpath":"4.9.3",
+        "emb.kotlin_on_classpath":"1.4.32",
+        "emb.is_emulator":"__EMBRACE_TEST_IGNORE__",
+        "emb.usage.add_breadcrumb":"1",
+        "emb.usage.add_user_persona":"1",
+        "emb.usage.set_user_email":"1",
+        "emb.usage.set_user_identifier":"1",
+        "emb.usage.set_username":"1"
       },
-      "end_time_unix_nano": "__EMBRACE_TEST_IGNORE__",
-      "events": [],
-      "name": "emb-session",
-      "parent_span_id": "0000000000000000",
-      "span_id": "__EMBRACE_TEST_IGNORE__",
-      "start_time_unix_nano": "__EMBRACE_TEST_IGNORE__",
-      "status": "OK",
-      "trace_id": "__EMBRACE_TEST_IGNORE__"
+      "end_time_unix_nano":"__EMBRACE_TEST_IGNORE__",
+      "events":[
+
+      ],
+      "name":"emb-session",
+      "parent_span_id":"0000000000000000",
+      "span_id":"__EMBRACE_TEST_IGNORE__",
+      "start_time_unix_nano":"__EMBRACE_TEST_IGNORE__",
+      "status":"OK",
+      "trace_id":"__EMBRACE_TEST_IGNORE__"
     }
   ],
-  "v": 13
-}
+  "v":13,
+  "resource":{
+
+  },
+  "metadata":{
+    "user_id":"some id",
+    "email":"user@email.com",
+    "username":"John Doe",
+    "personas":"__EMBRACE_TEST_IGNORE__",
+    "timezone_description":"__EMBRACE_TEST_IGNORE__",
+    "locale":"__EMBRACE_TEST_IGNORE__"
+  },
+  "version":"0.1.0",
+  "type":"session",
+  "data":{
+    "spans":[
```

Comment:
@bidetofevil I spotted that the number of spans is different compared to the legacy payload. I think this is probably related to setting it in 2 places in the same payload. Do you think it's reasonable to only set it in 1 place, or perhaps to extract the flushing of spans (and any similar data) to one place?